int8 max_interaction_num
